pkg/controller/framework/shared_informer.go

```diff
@@ -78,12 +77,30 @@ func NewSharedInformer(lw cache.ListerWatcher, objType runtime.Object, resyncPer
 	return sharedInformer
 }
 
-type sharedInformer struct {
-	store      cache.Store
+// NewSharedIndexInformer creates a new instance for the listwatcher.
+// TODO: create a cache/factory of these at a higher level for the list all, watch all of a given resource that can
+// be shared amongst all consumers.
+func NewSharedIndexInformer(lw cache.ListerWatcher, objType runtime.Object, resyncPeriod time.Duration, indexers cache.Indexers) SharedIndexInformer {
+	sharedIndexInformer := &sharedIndexInformer{
+		processor:        &sharedProcessor{},
+		indexer:          cache.NewIndexer(DeletionHandlingMetaNamespaceKeyFunc, indexers),
+		listerWatcher:    lw,
+		objectType:       objType,
+		fullResyncPeriod: resyncPeriod,
+	}
+	return sharedIndexInformer
```

Comment:
Can you please be consistent with NewSharedInformer (i.e. it creates controller).